pkg/pubsub/system_test.go

```diff
@@ -25,3 +29,139 @@ func TestNewSystem(t *testing.T) {
 		})
 	}
 }
+
+func TestSystem_UpsertConnection(t *testing.T) {
+	type fields struct {
+		connections map[string]connection.Connection
+		providers   map[string]string
+	}
+	type args struct {
+		ctx      context.Context
+		config   interface{}
+		name     string
+		provider string
+	}
+	tests := []struct {
+		name    string
+		fields  fields
+		args    args
+		wantErr bool
+	}{
+		{
+			name: "Updating connections for unsupported provider",
+			fields: fields{
+				connections: map[string]connection.Connection{},
+				providers:   map[string]string{"audit": "dapr"},
+			},
+			args: args{
+				ctx:      context.Background(),
+				config:   nil,
+				name:     "audit",
+				provider: "test",
+			},
+			wantErr: false,
```

Comment:
If the provider is unsupported, why are we not expecting an error?

Reply:
Ohh, my bad, returning error and handling it better in audit now